atlasdb-client/src/main/java/com/palantir/atlasdb/tracing/Tracing.java

```diff
@@ -21,16 +21,26 @@
 import com.google.errorprone.annotations.CompileTimeConstant;
 import com.palantir.atlasdb.keyvalue.api.TableReference;
 import com.palantir.atlasdb.logging.LoggingArgs;
+import com.palantir.logsafe.SafeArg;
 import com.palantir.tracing.CloseableTracer;
 import com.palantir.tracing.TagTranslator;
 import java.util.Collection;
 import java.util.Map;
 import java.util.function.BiConsumer;
 import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public interface Tracing {
+    Logger log = LoggerFactory.getLogger(Tracing.class);
 
     static CloseableTracer startLocalTrace(@CompileTimeConstant String operation, Consumer<TagConsumer> tagTranslator) {
+        CloseableTracer tracer = CloseableTracer.startSpan(operation, FunctionalTagTranslator.INSTANCE, tagTranslator);
+        log.info(
+                "startLocalTrace",
+                SafeArg.of("operation", operation),
+                SafeArg.of("tracer class", tracer.getClass()),
+                SafeArg.of("trace", tracer.toString()));
         return CloseableTracer.startSpan(operation, FunctionalTagTranslator.INSTANCE, tagTranslator);
```

Comment:
It would be cool to basically figure out if traceId when open == traceId on close? CloseableTracer is final, so can't override that, maybe you can log in a finally block? You'd need to check the ordering between finally and calling close on the try-with-resources.

Reply:
`A try-with-resources statement can have catch and finally blocks just like an ordinary try statement. In a try-with-resources statement, any catch or finally block is run after the resources declared have been closed.`
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html